src/events/mouse.js

```diff
@@ -929,4 +972,81 @@ p5.prototype._onwheel = function(e) {
   }
 };
 
+/**
+ * <p>The function <a href="#/p5/requestPointerLock">requestPointerLock()</a>
+ * locks the pointer to its current position and makes it invisible.
+ * Use <a href="#/p5/movedX">movedX</a> and <a href="#/p5/movedY">movedY</a> to get the difference the mouse was moved since
+ * the last call of draw</p>
+ * <p>Note that not all browsers support this feature</p>
+ * <p>This enables you to create experiences that aren't limited by the mouse moving out of the screen
+ * even if it is repeatedly moved into one direction. </p>
+ * <p>For example a first person perspective experience</p>
+ *
+ * @method requestPointerLock
+ * @example
+ * <div class="notest">
+ * <code>
+ * //click the canvas to lock the pointer
+ * //press ESC to exit lock mode
+ * function draw() {
+ *   background(237, 34, 93);
+ * }
+ * function mouseClicked() {
+ *   requestPointerLock();
+ * }
+ * </code>
+ * </div>
+ *
+ * @alt
+ * black 50x50 rect moves up and down with mouse move while no cursor is visible
+ *
+ */
+p5.prototype.requestPointerLock = function() {
+  // pointer lock object forking for cross browser
+  var canvas = this._curElement.elt;
+  canvas.requestPointerLock =
+    canvas.requestPointerLock || canvas.mozRequestPointerLock;
+  if (!canvas.requestPointerLock) {
+    console.log('requestPointerLock is not implemented in this browser');
+    return false;
+  }
+  canvas.requestPointerLock();
+  return true;
+};
+
+/**
+ * <p>The function <a href="#/p5/exitPointerLock">exitPointerLock()</a>
+ * exits a previously triggered <a href="#/p5/requestPointerLock">pointer Lock</a> programmatically
+ * for example to make ui elements usable etc
+ *
+ * @method exitPointerLock
+ * @example
+ * <div class="notest">
+ * <code>
+ * //click the canvas to lock the pointer
+ * //click again to exit (otherwise escape)
+ * let locked = false;
+ * function draw() {
+ *   background(237, 34, 93);
+ * }
+ * function mouseClicked() {
```

Comment:
Make these examples (request/exitPointerLock) show visible change? Maybe an opportunity to demonstrate a simple version of a first person experience?

Reply:
that might be a lot of code just saying

Reply:
It might be but I don't think it has to be. [Here is a link that has it down to about 12 lines of code](https://editor.p5js.org/stalgiag/sketches/grDsFZPNR), which I don't think is outside the realm of the acceptable if this is a primary perk of the feature. (Note that this example uses a build from an older version of this PR thus the deltaX/deltaY)

Reply:
https://github.com/jonnytest1/p5.js/tree/examplefirstpersonexperience how about this one ? 
@requestPointerLock

Reply:
That is definitely cooler but too complicated for a reference example. The example that I posted is much much simpler. You can modify it to match your sensibilities and style but I'd try to keep it under 20 lines.

Reply:
> It might be but I don't think it has to be. [Here is a link that has it down to about 12 lines of code](https://editor.p5js.org/stalgiag/sketches/grDsFZPNR), which I don't think is outside the realm of the acceptable if this is a primary perk of the feature. (Note that this example uses a build from an older version of this PR thus the deltaX/deltaY)

stolen :P